### PR TITLE
ZCS-5708 Handling of dynamic group

### PR DESCRIPTION
**Pull request by rupalid**
Added code to handle case when id of a dynamic group is passed.
Added code when an invalid id is passed for a group.
Verified the test cases specified in test rail.
Removed code which moved the group to the new oU
Added check to not allow moving groups across domain and ou

### CODE REVIEW

File: store/src/java/com/zimbra/cs/service/admin/ModifyHABGroup.java [hunk 1]
```diff
@@ -93,39 +103,64 @@ public void handleHABGroupMove(Provisioning prov, Element operation, Element res
...
-        LdapDistributionList group = (LdapDistributionList) prov
-            .getGroup(Key.DistributionListBy.id, habGroupId, true, false);
-        LdapDistributionList targetGroup = (LdapDistributionList) prov
-            .getGroup(Key.DistributionListBy.id, habTargetParentId, true, false);
-        LdapDistributionList currentGroup = (LdapDistributionList) prov
+
+        Group group = prov.getGroup(Key.DistributionListBy.id, habGroupId, true, false);
+        Group targetGroup = prov.getGroup(Key.DistributionListBy.id, habTargetParentId, true,
+            false);
+        LdapDistributionList currentList = (LdapDistributionList) prov
```
Comment (shrikant-prasad) [inline store/src/java/com/zimbra/cs/service/admin/ModifyHABGroup.java]: current list should also be a Group.

File: store/src/java/com/zimbra/cs/service/admin/ModifyHABGroup.java [hunk 2]
```diff
@@ -93,40 +114,109 @@ public void handleHABGroupMove(Provisioning prov, Element operation, Element res
...
+        Set<String> membersSet = targetGroup.getAllMembersSet();
+        for (String s : membersSet) {
+            membersEl.addElement(AdminConstants.E_MEMBER).addText(s);
+        }
+    }
+
+    /**
+     * @param targetGroup
+     * @param group
+     * @param currentGroup
```
Comment (chinmay15) [inline store/src/java/com/zimbra/cs/service/admin/ModifyHABGroup.java]: add all params

File: store/src/java/com/zimbra/cs/service/admin/ModifyHABGroup.java [hunk 3]
```diff
@@ -93,40 +114,109 @@ public void handleHABGroupMove(Provisioning prov, Element operation, Element res
...
+                throw ServiceException.NOT_FOUND(String.format("Error finding dn for group:%s", group.getName()));
+            }
+        }
+
+        LdapDistributionList targetList = (LdapDistributionList) targetGroup;
+        String parentDn = targetList.getDN();
+        parentDn = parentDn.substring(parentDn.indexOf(OU));
+        String targetDn = CN + cn + "," + parentDn;
+        ZimbraLog.misc.info("Target parent dn:%s and group dn:%s", targetDn, groupDn);
         prov.changeHABGroupParent(groupDn, targetDn);
```
Comment (shrikant-prasad) [inline store/src/java/com/zimbra/cs/service/admin/ModifyHABGroup.java]: we have to remove this. should not change ou of group.

File: store/src/java/com/zimbra/cs/service/admin/ModifyHABGroup.java [hunk 4]
```diff
@@ -93,40 +110,101 @@ public void handleHABGroupMove(Provisioning prov, Element operation, Element res
...
+            dlsToCheck.add(targetGroup.getMail());
+           if (prov.dlIsInDynamicHABGroup((DynamicGroup)group, dlsToCheck)) {
+               throw ServiceException.INVALID_REQUEST
+               (String.format("Target group:%s is a member of the group:%s being moved", targetGroup.getName(), group.getName()), null);
+           }
+        } else {
+            Set<String> members = group.getAllMembersSet();
+            for (String member : members) {
+                if (member.equalsIgnoreCase(targetGroup.getMail())) {
+                    throw ServiceException.INVALID_REQUEST(String.format("Group to be moved is a member of the target group"),
```
Comment (shrikant-prasad) [inline store/src/java/com/zimbra/cs/service/admin/ModifyHABGroup.java]: error msg should be "target group is member of group being moved"

**Pull request merged**